subworkflows/local/binning.nf

```diff
@@ -136,63 +120,20 @@ workflow BINNING {
 
     GUNZIP_BINS ( ch_final_bins_for_gunzip )
     ch_binning_results_gunzipped = GUNZIP_BINS.out.gunzip
-    ch_versions = ch_versions.mix(GUNZIP_BINS.out.versions.first())
+        .groupTuple(by: 0)
 
     GUNZIP_UNBINS ( ch_split_fasta_results_transposed )
     ch_splitfasta_results_gunzipped = GUNZIP_UNBINS.out.gunzip
-    ch_versions = ch_versions.mix(GUNZIP_UNBINS.out.versions.first())
-
-    // Compute bin depths for different samples (according to `binning_map_mode`)
-    // Have to remove binner meta before joining with according depths files,
-    // as required for MAG_DEPTHS, but we can add  'binner'
-    // info again based on file name and finally group by
-    // 'assembler', 'id', 'binner'
-    ch_depth_input = ch_binning_results_gunzipped
-        .mix(ch_splitfasta_results_gunzipped )
-        .map { meta, bin ->
-            def meta_new = meta.clone()
-            meta_new.remove('binner')
-            [ meta_new, bin ]
-        }
-        .groupTuple (by: 0 )
-        .join( METABAT2_JGISUMMARIZEBAMCONTIGDEPTHS.out.depth, by: 0 )
-        .transpose()
-        .map { meta, bin, contig_depths_file ->
-            def meta_new = meta.clone()
-            meta_new['binner'] = bin.name.split("-")[1]
-            [ meta_new, bin, contig_depths_file ]
-        }
-        .groupTuple (by: [0,2] )
-
-    MAG_DEPTHS ( ch_depth_input )
-    ch_versions = ch_versions.mix(MAG_DEPTHS.out.versions)
-
-    // Plot bin depths heatmap for each assembly and mapped samples (according to `binning_map_mode`)
-    // create file containing group information for all samples
-    ch_sample_groups = reads
-        .collectFile(name:'sample_groups.tsv'){ meta, reads -> meta.id + '\t' + meta.group + '\n' }
+        .groupTuple(by: 0)
```

Comment:
Do you think using `size` or `groupKey`will make this grouping operation more efficient?

Reply:
see https://github.com/nf-core/mag/pull/499#discussion_r1326993376